packages/@aws-cdk/aws-cloudfront/test/distribution.test.ts

```diff
@@ -286,6 +287,90 @@ describe('custom error responses', () => {
 
 });
 
+describe('with Lambda@Edge functions', () => {
+  let lambdaFunction: lambda.Function;
+  let origin: Origin;
+
+  beforeEach(() => {
+    lambdaFunction = new lambda.Function(stack, 'Function', {
+      runtime: lambda.Runtime.NODEJS,
+      code: lambda.Code.fromInline('whatever'),
+      handler: 'index.handler',
+    });
+
+    origin = Origin.fromBucket(s3.Bucket.fromBucketName(stack, 'Bucket', 'my-bucket'));
+  });
+
+  test('can add an edge lambdas to the default behavior', () => {
+    new Distribution(stack, 'MyDist', {
+      defaultBehavior: {
+        origin,
+        edgeLambdas: [
+          {
+            lambdaFunction: lambdaFunction.latestVersion,
+            eventType: LambdaEdgeEventType.VIEWER_REQUEST,
+          },
+        ],
+      },
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CloudFront::Distribution', {
+      DistributionConfig: {
+        DefaultCacheBehavior: {
+          LambdaFunctionAssociations: [
+            {
+              EventType: 'viewer-request',
+              LambdaFunctionARN: {
+                'Fn::Join': ['',
+                  [
+                    { 'Fn::GetAtt': ['Function76856677', 'Arn'] },
+                    ':$LATEST',
```

Comment:
As called out above, I believe (from the docs) that this won't deploy.

Reply:
Done in https://github.com/aws/aws-cdk/pull/9220#discussion_r459800775